core/main/src/broker/endpoint_broker.rs

```diff
@@ -696,13 +698,29 @@ impl BrokerOutputForwarder {
 
                         let request_id = rpc_request.ctx.call_id;
                         response.id = Some(request_id);
-
+                        let tm_str = get_rpc_header(&rpc_request);
                         // Step 2: Create the message
-                        let message = ApiMessage {
-                            request_id: request_id.to_string(),
-                            protocol: rpc_request.ctx.protocol.clone(),
-                            jsonrpc_msg: serde_json::to_string(&response).unwrap(),
-                        };
+                        let mut message = ApiMessage::new(
+                            rpc_request.ctx.protocol.clone(),
+                            serde_json::to_string(&response).unwrap(),
```

Comment:
unsafe unwrap.

Reply:
Existing code and its Serialization of an existing rust object. Even the underlying Serde impl only returns OK

```
#[inline]
pub fn to_string<T>(value: &T) -> Result<String>
where
    T: ?Sized + Serialize,
{
    let vec = tri!(to_vec(value));
    let string = unsafe {
        // We do not emit invalid UTF-8.
        String::from_utf8_unchecked(vec)
    };
    Ok(string)
}